File: 0009-palindrome-number/0009-palindrome-number.js
/**
 * @param {number} x
 * @return {boolean}
 */
var isPalindrome = function(x) {
    let str = String(x);
    let left = 0;
    let right = str.length - 1;
    
    while (right >= left) {
        if (str[left] !== str[right]) {
            return false;
        }
        
        left++;
        right--
    }
    
    
    return true;
};

File: 0009-palindrome-number/0009-palindrome-number.py
class Solution:
    def isPalindrome(self, x: int) -> bool:
        NumStr = str(x)
        length = len(NumStr)
        left = 0;
        right = length - 1
        
        while left <= right:
            if NumStr[left] != NumStr[right]:
                return False
            left+=1
            right-=1
        
        return True